File: WebML/src/main/java/Points/Point.java
package Points;

public class Point{
    private int X;
    private int Y;
    public int value;

    public int[] getCartesian(){
        return new int[]{X, Y};
    }

    double[] getPolar(){
        double ro = X*X + Y*Y;
        double phi = Math.atan2(Y, X);
        return new double[]{Math.sqrt(ro), phi};
    }

    /**
      * accepts cartesian coordinates as params
     */
    Point(int x, int y, int val){
        X = x;
        Y = y;
        value = val;
    }

    /**
     * accepts polar coordinates of points
     * @param ro distance to the zero of axis
     * @param fi angle
     * @param val value of point
     */
    Point(double ro, double fi, int val){
        X = (int) (ro * Math.sin(fi));
        Y = (int) (ro * Math.cos(fi));
        value = val;
    }

    void setX(int X){
        this.X = X;
    }

    void setY(int Y){
        this.Y = Y;
    }

    void setValue(int value){
        this.value = value;
    }

    Double distanceTo(Point other){
        return Math.sqrt((this.X - other.X) * (this.X - other.X) +
                (this.Y - other.Y) * (this.Y - other.Y));
    }
}

File: Server/Points/Point.py
from math import cos, sin, atan2, sqrt


class Point:
    def get_cartesian(self):
        return self.X, self.Y

    def get_polar(self):
        ro = sqrt(self.X ** 2 + self.Y ** 2)
        phi = atan2(self.Y, self.X)
        return ro, phi

    def __init__(self, val, x=None, y=None, ro=None, phi=None):
        self.val = val
        if x is not None and y is not None:
            self.X = x
            self.Y = y
        else:
            self.X = ro*cos(phi)
            self.Y = ro*sin(phi)

    def distance_to(self, other):
        if not hasattr(other, 'X'):
            return sqrt((self.X - other[0])**2 + (self.Y-other[1])**2)
        else:
            return sqrt((self.X - other.X) ** 2 + (self.Y - other.Y) ** 2)

    def __str__(self):
        return "{}, {}, {}\n".format(self.X, self.Y, self.val)
